packages/loading-screen/src/main.js

```diff
@@ -0,0 +1,28 @@
+import Vue from 'vue';
+import { BootstrapVue, IconsPlugin } from 'bootstrap-vue';
+import App from '@/App.vue';
+
+Vue.use(BootstrapVue);
+Vue.use(IconsPlugin);
+
+Vue.config.productionTip = false;
+
+window.app = new Vue({
+  render: (h) => h(App, { ref: 'App' }),
+}).$mount('#app');
+
+window.show_retry = () => {
+  window.app.$refs.App.setState('retry');
+}
```

Comment:
These assignements are hacky. Can't the callbacks be hooked when the App is mounted? Or in any other of the [lifecycle hooks](https://v3.vuejs.org/api/options-lifecycle-hooks.html)?